daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -36,9 +38,9 @@ object TransactionCoder {
     * @tparam Val value type
     * @return protobuf wire format contract instance
     */
-  def encodeContractInstance[Val](
+  def encodeContractInstance[Val <: Value.VersionedValue[Value.ContractId]](
```

Comment:
At this point there is no need for having the type parameter `Val` anymore.

Reply:
good catch !